File: 11-Container_with_most_water/2ptrs.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        bestContainer = 0
        p1 = 0
        p2 = len(height) - 1

        while p1 < p2:

            w = p2 - p1
            h = min(height[p1], height[p2])
            area = w * h

            bestContainer = max(area, bestContainer)

            if height[p1] < height[p2]:
                p1 += 1
            else:
                p2 -= 1

        return bestContainer

File: 11-Container_with_most_water/2ptrs.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut bestcontainer = 0;
        let mut p1 = 0;
        let mut p2 = height.len() - 1;

        while p1 < p2 {
            let w = p2 - p1;
            let h = height[p1].min(height[p2]);
            let area = w * h as usize;

            bestcontainer = bestcontainer.max(area as i32);

            if height[p1] < height[p2] {
                p1 += 1;
            } else {
                p2 -= 1;
            }
        }

        bestcontainer
    }
}
